RadClass/RadClass.py

```diff
@@ -0,0 +1,204 @@
+        each bin.
+
+        Return: 1D numpy array for integration interval
+        '''
+
+        # extract requisite data rows
+        data_matrix = self.processor.data_slice(self.datapath, rows)
+
+        # normalize by live times to produce count rate data
+        # processor.live can be indexed by appropriate timestamps but
+        # data_matrix only has indices for rows.
+        data_matrix = data_matrix / self.processor.live[rows][:, None]
+
+        # utilizes numpy architecture to sum data
+        total = np.sum(data_matrix, axis=0)
+
+        return total
+
+    def collect_rows(self):
+        '''
+        Uses the integration duration to collect all rows needed for analysis.
+        Uses numpy.arange() to enumerate row indices from start to end.
+
+        Return: list of rows for data indexing
+        '''
+
+        # collect start index from tracked timestamp
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+
+        # NOTE: this behavior is currently disabled by the if-statement in
+        # march() removing the 'or' portion with allow this to work
+        #
+        # if the final portion of the file is smaller than a full integration
+        # interval, only what is left is collected for this analysis
+        end_i = min(start_i + self.integration,
+                    len(self.processor.timestamps) - 1)
+
+        # enumerate number of rows to integrate exclusive of the endpoint
+        rows = np.arange(start_i, end_i)
+        return rows
+
+    def march(self):
+        '''
+        Advance a step using stride time to find the new working integration
+        interval. Also keeps track of the EOF state but checking for
+        out-of-index situations.
+
+        Returns: boolean for EOF
+        '''
+
+        # find the working interval starting index and advance stride
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+        new_i = start_i[0] + self.stride
+
+        # stop analysis if EOF reached
+        # NOTE: stops prematurely, for windows of full integration only
+        running = True
+        if (new_i >= len(self.processor.timestamps)) or ((new_i + self.integration) >= len(self.processor.timestamps)):
+            running = False
+
+        if running:
+            # update working integration interval timestep
+            self.working_time = self.processor.timestamps[new_i]
+
+        return running
+
+    def iterate(self):
+        '''
+        Full iteration over the entirety of the data file. Runs
+        until EOF reached. Prints progress over the course of the analysis.
+        Only runs for a set node (datapath) with data already queued.
+        '''
+        bar = progressbar.ProgressBar(max_value=100, redirect_stdout=True)
+        inverse_dt = 1.0 / (self.processor.timestamps[-1] - self.processor.timestamps[0])
+
+        log_interval = 10000  # number of samples analyzed between log updates
```

Comment:
I made this magic number into a variable but I'm not sure where it should live. It could stay in `iterate()` since that is the only place it is used or it could be a member variable. Thoughts?

Reply:
Probably fine as a local variable